src/modbus-rtu.c

```diff
@@ -1180,8 +1180,16 @@ static int _modbus_rtu_select(modbus_t *ctx, fd_set *rset,
 }
 
 static void _modbus_rtu_free(modbus_t *ctx) {
-    free(((modbus_rtu_t*)ctx->backend_data)->device);
-    free(ctx->backend_data);
+    if (ctx->backend_data) {
+        modbus_rtu_t *ctx_rtu = (modbus_rtu_t *)ctx->backend_data;
+
+        if (ctx_rtu->device) {
+            free(ctx_rtu->device);
+        }
```

Comment:
free(NULL) is defined as NOP, so this check is unnecessary. Good check on the outer deref though!